src/table.ts

```diff
@@ -1373,13 +1373,16 @@ class Table extends common.ServiceObject {
       metadata.schema = Table.createSchemaFromString_(metadata.schema);
     }
 
-    extend(true, metadata, {
-      destinationTable: {
-        projectId: this.bigQuery.projectId,
-        datasetId: this.dataset.id,
-        tableId: this.id,
+    metadata = {
```

Comment:
I don't believe this style will do a deep clone of the object:

```js
const defaultConfig = { deep: { obj: { value: false }}}
const userConfig = { deep: { obj: { value: true }}}

// Override defaults with user config
const metadata = {...defaultConfig, ...userConfig}

// Confirm the overwrite worked...
console.log(metadata.deep.obj.value) // true (good!)

// Set a new property on the clone object...
metadata.deep.obj.newValue = false

// Check if it modified the user's config object
console.log(userConfig.deep.obj.newValue) // false (d'oh! The user's object should not be modified)
```

A fix for this would be sticking to `extend`:

```diff
// Override defaults with user config
-const metadata = {...defaultConfig, ...userConfig}
+const metadata = extend(true, defaultConfig, userConfig)
```

Reply:
@stephenplusplus Thanks!! I was using same solution that you provided but was not reassign to `metadata` so it wasn't extend the existing object. but now it's worked.

Reply:
Thank you!